src/include/btmem.h

```diff
@@ -1114,8 +1114,12 @@ struct __wt_update {
      */
     volatile uint8_t prepare_state; /* prepare state */
 
-    /* Temp: Remove after integrating las cursors with btree cursors */
-    uint8_t ext;
+/* AUTOMATIC FLAG VALUE GENERATION START */
+#define WT_UPDATE_HISTORY_STORE 0x1u      /* Update has been written to history store. */
+#define WT_UPDATE_RESTORED_FROM_DISK 0x2u /* Update is temporary retrieved from LAS. */
+                                          /* AUTOMATIC FLAG VALUE GENERATION STOP */
+    uint8_t flags;
+    uint8_t unused; /* Unused value for structure padding. */
```

Comment:
Why do we need structure padding?

Reply:
@hbokhari suggested this change as it pads out the structure to be a multiple of 4 bytes. Could you comment on this @hbokhari?

Reply:
@kommiharibabu suggested this change as it pads the structure out to be a multiple of 4 bytes. Could you answer Keith's question, @kommiharibabu?

Reply:
The change was suggested to use a 1-byte flag and a 1-byte used variable to make sure that `upd->data` variable gets aligned to the multiple of 4 bytes.

Reply:
@kommiharibabu, why do you want `upd.data` to be aligned to a multiple of 4 bytes?

Reply:
During the discussion of changing the `flags` variable from `uint32_t` to `uint8_t`, the padding discussion arose as `upd.data` is not aligned while discussing it with Vamsi, so the suggestion of adding `unused` variable has popped up. 

In the code, `upd.data` is mostly used to assign the address. Accessing the `upd.data` needs more instructions if it is not aligned?

Reply:
Is it any slower to access a single byte if the address isn't aligned? Since we're not doing multi-byte instructions, does it matter?

Reply:
@keithbostic, Thanks for the explanation.
Most of the places where the `upd.data` is passed between the functions, it is passed via `void *` pointer, so there is no access to the data. And cases where there is a need for data access, the data is `memcpy` into the aligned address variable before it is accessed.

>    
    /* Get the number of modify entries. */
    p = modify;
    memcpy(&tmp, p++, sizeof(size_t));
    nentries = (int)tmp;

The above code in `__wt_modify_idempotent()` function, where the `upd.data` address accessed by copying into a alinged variable.

Already the existing code in the `WiredTiger` already taken care of accessing the `upd.data` as an unaligned address whenever there is a need. So there is no need of doing structure padding that increases the memory size.